api/libproject/devstart.rb

```diff
@@ -415,6 +415,48 @@ def run
   end
 end
 
+
+# Command-line parsing and main "run" implementation for deploy-api.
+class DeployApi < GcloudContext
+  # Adds command-line flags specific to set-authority.
+  def add_options
+    super
+    @parser.on(
+        "--version [VERSION]",
+        "The name of the version to deploy. Required."
+        ) do |version|
+      @opts.version = version
+    end
+    @parser.on(
+        "--promote",
+        "Use this if you want to promote this version so it receives traffic. By default, it won't."
+        ) do |promote|
+      @opts.promote = "promote"
+    end
+    @opts.promote = "no-promote" # default
+  end
+
+  def validate_options
+    super
+    if @opts.version == nil
+      puts @parser.help
+      exit 1
+    end
+  end
+
+  def run
+    super do
+      read_db_vars(@opts.creds_file, @opts.project)
+      # This triggers logic in generate_appengine_web_xml.sh to use environment variables set above.
+      ENV["CIRCLECI"] = "true"
```

Comment:
I think this happens:
 1. `read_db_vars` downloads a project-specific file, and copies it to `vars.env`.
 2. `read_db_vars` reads the contents of that `vars.env` into `ENV`.
 3. We set `CIRCLECI=true` here.
 4. `generate_appengine_web_xml.sh` sees `CIRCLECI=true` and reads `vars.env`.
If you omit the `ENV["CIRCLECI"] = "true"` line here, does whatever context that runs `generate_appengine_web_xml.sh` pick up the `ENV` modifications? (It sees `CIRCLECI`, so I would assume so?)

Reply:
Not quite. It's:

1. read_db_vars downloads a project specific file, and sets environment variables based on the contents.
2. CIRCLECI=true tells generate_appengine_web_xml.sh to use environment variables, rather than vars.env, for the DB connection information.

Updated the comments.